src/averageAges.js

```diff
@@ -36,8 +46,17 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
+
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const womanFilter = people.filter(person => {
+    return person.sex === 'f'
+    && (withChildren === undefined
+      ? true : people.some(man => man.mother === person.name));
+  });
```

Comment:
its to complicate , could we made its simplier ?) cause its hard to understand what happened here, especially if you are not in context